### ISSUE TITLE
Observables does not send updated value when app enters foreground

### ISSUE THREAD

**Issue by havard024** [RxSwiftCommunity/RxCoreLocation#24]
Our app requires the user to allow access to the devices location. 
To handle this we first check if the location service is enabled, if it's not enabled we present a view controller letting the user know that he / she needs to enable the location service to continue.
The user would then most likely put the app in the background, enable the location service, open the app again.

In the view we present to the user when location service is disabled, we would like to automatically detect if the location service has been enabled, the code below is our attempt to react to the `isEnabled` event.

```swift
viewDidLoad() {
  super.viewDidLoad()
  manager.rx
    .isEnabled
    .debug("isEnabled")
    .subscribe(onNext: { isEnabled in
      if isEnabled {
        DispatchQueue.main.async {
          self.dismiss(animated: true, completion: nil)
        }
      }
    })
    .disposed(by: bag)
}
```

When we test this by putting the app in the background then to foreground, no event is triggered.

Is this something the library is designed to handle? 

If not, how would one go about creating this functionality in an app?

**Comment by bobgodwinx** [comment]
By the way this is the wrong way to do subscription. you should use `subscribeOn(MainScheduler.instance)`  instead. 
And you are also missing something before the call of `manager.rx...`      

```Swift
        manager.requestWhenInUseAuthorization()
        manager.startUpdatingLocation()
```

[please see the example project for more](https://github.com/RxSwiftCommunity/RxCoreLocation/blob/master/RxCoreLocationExample/ViewController.swift#L24)

**Comment by unxavi** [comment]
Hi,

Why do you need to startUpdatingLocation to know if location services are on? this is a static method of CLLocationManager and shouldn't be related between them.

Using the example I project config, I also see that sometimes it's not deliver in background when you go to settings and change locations services on/off

**Comment by havard024** [comment]
@bobgodwinx I'm referring to the [example on line 85](https://github.com/RxSwiftCommunity/RxCoreLocation/blob/60d24645ea141d8c7156826f6140643311988be5/RxCoreLocationExample/ViewController.swift#L85)

```
manager.rx
    .isEnabled
    .debug("isEnabled")
    .subscribe(onNext: { _ in })
    .disposed(by: bag)
 ```

**Comment by bobgodwinx** [comment]
@havard024 I think I now understand what you're trying todo here. Correct me if I am wrong. 
1. You are putting the app in background and disactivating location services correct ?

**Comment by havard024** [comment]
@bobgodwinx Yes exactly and when you put the app in foreground again, there is no event to notify that the location service is disabled

**Comment by bobgodwinx** [comment]
@havard024  Here is my advice: The API is actually working as expected. The event you are expecting is not emitted because `CLLocationManager` actually emitted `Error` events which you could use for this purpose.  

-  `didError -> Event next((manager: <CLLocationManager: 0x600003acbfb0>, error: Error Domain=kCLErrorDomain Code=1 "(null)"))` the code here is 1 which means `kCLErrorDenied`. 
- This error is actually indicating that the manager is locked out of `CLLocationManager`
-  would rather use another event here which is the `didChangeAuthorization ` that also gets emitted when you try switching of the `LocationServices` 

```swift
        manager.rx
            .didChangeAuthorization
            .debug("didChangeAuthorization")
            .subscribe(onNext: { e in
                let a = e.status
                switch a {
                case .denied, .notDetermined, .restricted:
                    print("Stay there blocking the screen! don't move")
                default:
                    print("Stop blocking the screen let him enter now")
                }
            })
            .disposed(by: bag)
```
Let me know if this helps.
